src/Uno.UI.RuntimeTests/MUX/Microsoft_UI_Xaml_Controls/TreeViewTests.cs

```diff
@@ -0,0 +1,820 @@
+			{
+				treeView = new TreeView { SelectionMode = TreeViewSelectionMode.Multiple };
+				treeView.SelectionChanged += (s, e) => selectionChangedEventArgs = e;
+
+				node1 = new TreeViewNode { Content = "1", IsExpanded = true };
+				node11 = new TreeViewNode { Content = "11" };
+				node12 = new TreeViewNode { Content = "12" };
+				node13 = new TreeViewNode { Content = "13" };
+				node1.Children.Add(node11);
+				node1.Children.Add(node12);
+				node1.Children.Add(node13);
+
+				node2 = new TreeViewNode { Content = "2", IsExpanded = true };
+				node21 = new TreeViewNode { Content = "21" };
+				node2.Children.Add(node21);
+
+				node3 = new TreeViewNode { Content = "3" };
+
+				treeView.RootNodes.Add(node1);
+				treeView.RootNodes.Add(node2);
+				treeView.RootNodes.Add(node3);
+				TestServices.WindowHelper.WindowContent = treeView;
+			});
+
+			await TestServices.WindowHelper.WaitForIdle();
+
+			await RunOnUIThread.Execute(() =>
+			{
+				var tvi1 = (TreeViewItem)treeView.ContainerFromItem(node1);
+				var tvi11 = (TreeViewItem)treeView.ContainerFromItem(node11);
+				var tvi12 = (TreeViewItem)treeView.ContainerFromItem(node12);
+				var tvi13 = (TreeViewItem)treeView.ContainerFromItem(node13);
+				var tvi2 = (TreeViewItem)treeView.ContainerFromItem(node2);
+				var tvi21 = (TreeViewItem)treeView.ContainerFromItem(node21);
+				var tvi3 = (TreeViewItem)treeView.ContainerFromItem(node3);
+
+				// - 1         selected
+				//   - 11      selected
+				//   - 12      selected
+				//   - 13      selected
+				// - 2
+				//   - 21
+				// - 3
+				tvi1.IsSelected = true;
+
+				Verify.IsNotNull(selectionChangedEventArgs);
+				Verify.AreEqual(4, selectionChangedEventArgs.AddedItems.Count);
+				Verify.IsTrue(selectionChangedEventArgs.AddedItems.Contains(node1));
+				Verify.IsTrue(selectionChangedEventArgs.AddedItems.Contains(node11));
+				Verify.IsTrue(selectionChangedEventArgs.AddedItems.Contains(node12));
+				Verify.IsTrue(selectionChangedEventArgs.AddedItems.Contains(node13));
+				Verify.AreEqual(0, selectionChangedEventArgs.RemovedItems.Count);
+				selectionChangedEventArgs = null;
+
+				// - 1         selected
+				//   - 11
+				//   - 12      selected
+				//   - 13      selected
+				// - 2
+				//   - 21
+				// - 3
+				tvi11.IsSelected = true;
+				tvi11.IsSelected = false;
+
+				Verify.IsNotNull(selectionChangedEventArgs);
+				Verify.AreEqual(0, selectionChangedEventArgs.AddedItems.Count);
+				Verify.AreEqual(1, selectionChangedEventArgs.RemovedItems.Count);
+				Verify.IsTrue(selectionChangedEventArgs.RemovedItems.Contains(node11));
+				selectionChangedEventArgs = null;
+
+				// - 1         selected
+				//   - 11      selected
+				//   - 12      selected
+				//   - 13      selected
+				// - 2         selected
+				//   - 21      selected
+				// - 3         selected
+				treeView.SelectAll();
+				Verify.IsNotNull(selectionChangedEventArgs);
+				var items = selectionChangedEventArgs.AddedItems.ToList();
```

Comment:
Codacy found an issue: [Remove the unused local variable 'items'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)